### PR TITLE
minor cleanups

### PR DESCRIPTION
**Pull request by dfa1**
- using JDK methods when possible (Arrays.fill, Math.max, Objects.equal)
- diamond operator
- dandling javadoc

### CODE REVIEW

File: src/main/java/io/vertx/core/eventbus/impl/HandlerHolder.java [hunk 1]
```diff
@@ -54,7 +56,7 @@ public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     HandlerHolder that = (HandlerHolder) o;
-    if (handler != null ? !handler.equals(that.handler) : that.handler != null) return false;
+    if (!Objects.equals(handler, that.handler)) return false;
     return true;
```
Comment (julianladisch) [inline src/main/java/io/vertx/core/eventbus/impl/HandlerHolder.java:60]:
```suggestion
    return Objects.equals(handler, that.handler);
```

File: src/main/java/io/vertx/core/net/impl/SocketAddressImpl.java [hunk 2]
```diff
@@ -141,8 +141,8 @@ public boolean equals(Object o) {
     SocketAddressImpl that = (SocketAddressImpl) o;
 
     if (port != that.port) return false;
-    if (host != null ? !host.equals(that.host) : that.host != null) return false;
-    if (path != null ? !path.equals(that.path) : that.path != null) return false;
+    if (!Objects.equals(host, that.host)) return false;
+    if (!Objects.equals(path, that.path)) return false;
 
     return true;
```
Comment (julianladisch) [inline src/main/java/io/vertx/core/net/impl/SocketAddressImpl.java:147]:
```suggestion
    return Objects.equals(host, that.host) && Objects.equals(path, that.path);
```

File: src/main/java/io/vertx/core/net/impl/VertxEventLoopGroup.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ public boolean equals(Object o) {
 
       EventLoopHolder that = (EventLoopHolder) o;
 
-      if (worker != null ? !worker.equals(that.worker) : that.worker != null) return false;
+      if (!Objects.equals(worker, that.worker)) return false;
 
       return true;
```
Comment (julianladisch) [inline src/main/java/io/vertx/core/net/impl/VertxEventLoopGroup.java:160]:
```suggestion
      return Objects.equals(worker, that.worker);
```

**Comment by dfa1** [comment]
thank you @julianladisch  for your review. I tried to fix everything you mentioned, please have another look.

**Review by tsegismont — CHANGES_REQUESTED** [review]
@dfa1thanks, a few concerns

File: src/main/java/io/vertx/core/net/impl/ServerID.java [hunk 4]
```diff
@@ -32,21 +33,15 @@ public ServerID() {
...
+    ServerID that = (ServerID) o;
+    return port != that.port && Objects.equals(host, that.host);
   }
 
   @Override
   public int hashCode() {
-    int result = port;
-    result = 31 * result + host.hashCode();
-    return result;
+    return Objects.hash(port, host);
```
[the remark this answers is not in the document]
Comment (dfa1) [reply src/main/java/io/vertx/core/net/impl/ServerID.java]: thank you @tsegismont :) I just pushed the fix, please have another look

File: src/main/java/io/vertx/core/net/impl/SocketAddressImpl.java [hunk 5]
```diff
@@ -139,19 +139,11 @@ public boolean equals(Object o) {
...
+    return port != that.port && Objects.equals(host, that.host) && Objects.equals(path, that.path);
   }
 
   @Override
   public int hashCode() {
-    int result = host != null ? host.hashCode() : 0;
-    result = 31 * result + (path != null ? path.hashCode() : 0);
-    result = 31 * result + port;
-    return result;
+    return Objects.hash(port, host, path);
```
[the remark this answers is not in the document]
Comment (dfa1) [reply src/main/java/io/vertx/core/net/impl/SocketAddressImpl.java]: changed to use `Objects.hashCode` that is exactly:

    public static int hashCode(Object o) {
        return o != null ? o.hashCode() : 0;
    }

**Review by tsegismont — APPROVED** [review]
Thank you @dfa1 !

**Pull request merged**